main.py

```diff
@@ -1,255 +1,111 @@
 import argparse
 import logging
-from os import getenv
-import kubernetes
+import json
 import pika
 import wagglemsg
-from pathlib import Path
-import time
-from dataclasses import dataclass
-from pod_event_watcher import PluginPodEventWatcher
-import amqp
-from prometheus_client import start_http_server, Counter, Gauge
+
 from contextlib import ExitStack
+from os import getenv
+from functools import lru_cache
+from prometheus_client import start_http_server, Counter
+from redis import Redis
 
 
-wes_data_service_messages_handled_total = Counter("wes_data_service_messages_handled_total", "Total number of messages handled.")
-wes_data_service_messages_invalid_total = Counter("wes_data_service_messages_invalid_total", "Total number of invalid messages.")
-wes_data_service_messages_backlogged_total = Counter("wes_data_service_messages_backlogged_total", "Total number of messages which have been backlogged.")
-wes_data_service_messages_published_total = Counter("wes_data_service_messages_published_total", "Total number of messages published.")
-wes_data_service_messages_expired_total = Counter("wes_data_service_messages_expired_total", "Total number of messages expired.")
-wes_data_service_pods_expired_total = Counter("wes_data_service_pods_expired_total", "Total number of pods expired.")
-wes_data_service_messages_in_backlog = Gauge("wes_data_service_messages_in_backlog", "Number of messages currently in backlog.")
-wes_data_service_pods_in_backlog = Gauge("wes_data_service_pods_in_backlog", "Number of pods currently in backlog.")
+wes_data_service_messages_total = Counter("wes_data_service_messages_total", "Total number of messages handled.", [])
+wes_data_service_messages_rejected_total = Counter("wes_data_service_messages_rejected_total", "Total number of invalid messages.")
+wes_data_service_messages_published_node_total = Counter("wes_data_service_messages_published_node_total", "Total number of messages published to node.")
+wes_data_service_messages_published_beehive_total = Counter("wes_data_service_messages_published_beehive_total", "Total number of messages published to beehive.")
 
 
 class InvalidMessageError(Exception):
-    """InvalidMessageError is throw to indicate that a message is invalid. (Bad JSON data, missing fields, etc.)"""
-
     def __init__(self, error):
         self.error = error
 
 
-class MonotonicClock:
-
-    def now(self):
-        return time.monotonic()
-
+class AppMetaCache:
 
-@dataclass
-class PodState:
+    def __init__(self, host="wes-app-meta-cache"):
+        self.client = Redis(host)
 
-    pod: object
-    backlog: list
-    updated_at: float
-
-
-@dataclass
-class MessageHandlerConfig:
-    node: str
-    vsn: str
-    upload_publish_name: str
-    pod_state_expire_duration: float = 7200.0
-    pod_without_metadata_state_expire_duration: float = 300.0
+    @lru_cache(maxsize=128)
```

Comment:
This adds a small second, in-memory LRU cache on top of Redis. (I picked 128 because I generally didn't think we'd have more than 128 apps running at the same time.)

This allows even faster access to app meta by eliminating the need to even do the network call to Redis, if that app_uid is still in the AppMetaCache.

(It's part of the standard library in the functools module - you can find more [here](https://docs.python.org/3/library/functools.html#functools.lru_cache).)